clang/test/CIR/CodeGen/types-IEEE-quad.c

```diff
@@ -0,0 +1,20 @@
+// RUN: %clang_cc1 -triple aarch64-none-linux-android21 -emit-cir %s -o %t.cir
+// RUN: FileCheck --input-file=%t.cir --check-prefix=CHECK %s
+
+long double i = 0;
+long double t2(long double i2) {
+    return i2 + i ; 
+}
+
+// CHECK: cir.global external @i = #cir.fp<0.000000e+00> : !cir.long_double<!cir.f128> {alignment = 16 : i64} loc({{.*}})
+// CHECK: cir.func @t2(%arg0: !cir.long_double<!cir.f128> loc({{.*}})) -> !cir.long_double<!cir.f128>
+// CHECK: %{{[0-9]+}} = cir.alloca !cir.long_double<!cir.f128>, !cir.ptr<!cir.long_double<!cir.f128>>, ["i2", init] {alignment = 16 : i64}
+// CHECK: %{{[0-9]+}} = cir.alloca !cir.long_double<!cir.f128>, !cir.ptr<!cir.long_double<!cir.f128>>, ["__retval"] {alignment = 16 : i64}    
+// CHECK: cir.store %arg0, %{{[0-9]+}} : !cir.long_double<!cir.f128>, !cir.ptr<!cir.long_double<!cir.f128>>
+// CHECK: %{{[0-9]+}} = cir.load %{{[0-9]+}} : !cir.ptr<!cir.long_double<!cir.f128>>, !cir.long_double<!cir.f128>
+// CHECK: %{{[0-9]+}} = cir.get_global @i : !cir.ptr<!cir.long_double<!cir.f128>>
+// CHECK: %{{[0-9]+}} = cir.load %{{[0-9]+}} : !cir.ptr<!cir.long_double<!cir.f128>>, !cir.long_double<!cir.f128>
+// CHECK: cir.store %{{[0-9]+}}, %{{[0-9]+}} : !cir.long_double<!cir.f128>, !cir.ptr<!cir.long_double<!cir.f128>>
+// CHECK: %{{[0-9]+}} = cir.load %{{[0-9]+}} : !cir.ptr<!cir.long_double<!cir.f128>>, !cir.long_double<!cir.f128>
+// CHECK: cir.return %{{[0-9]+}} : !cir.long_double<!cir.f128>
+
```

Comment:
You don't generally need to test every single operation, but given that you're adding a new type, it's fair to check that all the types are what you'd expect. A couple of [FileCheck](https://llvm.org/docs/CommandGuide/FileCheck.html) features can be useful though:
* CHECK-NEXT to verify a match is on the next line.
* CHECK-LABEL to demarcate logical parts of the output (e.g. functions).
* Numeric substitution blocks to easily match variables and confirm that the variables being used in an operation are what you'd expect.

```suggestion
// CHECK-LABEL: cir.func @t2(%arg0: !cir.long_double<!cir.f128> loc({{.*}})) -> !cir.long_double<!cir.f128>
// CHECK-NEXT:    %[[#I2:]] = cir.alloca !cir.long_double<!cir.f128>, !cir.ptr<!cir.long_double<!cir.f128>>, ["i2", init] {alignment = 16 : i64}
// CHECK-NEXT:    %[[#RETVAL:]] = cir.alloca !cir.long_double<!cir.f128>, !cir.ptr<!cir.long_double<!cir.f128>>, ["__retval"] {alignment = 16 : i64}
// CHECK-NEXT:    cir.store %arg0, %[[#I2]] : !cir.long_double<!cir.f128>, !cir.ptr<!cir.long_double<!cir.f128>>
// CHECK-NEXT:    %[[#I2_LOAD:]] = cir.load %[[#I2]] : !cir.ptr<!cir.long_double<!cir.f128>>, !cir.long_double<!cir.f128>
// CHECK-NEXT:    %[[#I:]] = cir.get_global @i : !cir.ptr<!cir.long_double<!cir.f128>>
// CHECK-NEXT:    %[[#I_LOAD:]] = cir.load %[[#I]] : !cir.ptr<!cir.long_double<!cir.f128>>, !cir.long_double<!cir.f128>
// CHECK-NEXT:    %[[#ADD:]] = cir.binop(add, %[[#I2_LOAD]], %[[#I_LOAD]]) : !cir.long_double<!cir.f128>
// CHECK-NEXT:    cir.store %[[#ADD]], %[[#RETVAL]] : !cir.long_double<!cir.f128>, !cir.ptr<!cir.long_double<!cir.f128>>
// CHECK-NEXT:    %[[#RETVAL_LOAD:]] = cir.load %[[#RETVAL]] : !cir.ptr<!cir.long_double<!cir.f128>>, !cir.long_double<!cir.f128>
// CHECK-NEXT:    cir.return %[[#RETVAL_LOAD]] : !cir.long_double<!cir.f128>
```

Reply:
Here, instead of doing a simple regex match, we are using named variables to make the match more precise.
Got it! I will keep in mind to use the FileCheck feature from here on out, as you pointed out.